react/features/authentication/components/LoginDialog.native.js

```diff
@@ -0,0 +1,280 @@
+import React, { Component } from 'react';
+import { connect as reduxConnect } from 'react-redux';
+import {
+    Button,
+    Modal,
+    Text,
+    TextInput,
+    View
+} from 'react-native';
+import {
+    authenticateAndUpgradeRole,
+    cancelLogin
+} from '../actions';
+import {
+    connect,
+    toJid
+} from '../../base/connection';
+import { translate } from '../../base/i18n';
+import { JitsiConnectionErrors } from '../../base/lib-jitsi-meet';
+import styles from './styles';
+
+/**
+ * Dialog asks user for username and password.
+ *
+ * First authentication configuration that it will deal with is the main XMPP
+ * domain (config.hosts.domain) with password authentication. A LoginDialog
+ * will be opened after 'CONNECTION_FAILED' action with
+ * 'JitsiConnectionErrors.PASSWORD_REQUIRED' error. After username and password
+ * are entered a new 'connect' action from 'features/base/connection' will be
+ * triggered which will result in new XMPP connection. The conference will start
+ * if the credentials are correct.
+ *
+ * The second setup is the main XMPP domain with password plus guest domain with
+ * anonymous access configured under 'config.hosts.anonymousdomain'. In such
+ * case user connects from the anonymous domain, but if the room does not exist
+ * yet, Jicofo will not allow to start new conference. This will trigger
+ * 'CONFERENCE_FAILED' action with JitsiConferenceErrors.AUTHENTICATION_REQUIRED
+ * error and 'authRequired' value of 'features/base/conference' will hold
+ * the {@link JitsiConference} instance. If user decides to authenticate a new
+ * {@link JitsiAuthConnection} will be created from which separate XMPP
+ * connection is established and authentication is performed. In case it
+ * succeeds Jicofo will assign new session ID which then can be used from
+ * the anonymous domain connection to create and join the room. This part is
+ * done by {@link JitsiAuthConnection} from lib-jitsi-meet.
+ *
+ * See https://github.com/jitsi/jicofo#secure-domain for configuration
+ * parameters description.
+ */
+class LoginDialog extends Component {
+    /**
+     * LoginDialog component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * {@link JitsiConference} that needs authentication - will hold a valid
+         * value in XMPP login + guest access mode.
+         */
+        conference: React.PropTypes.object,
```

Comment:
PropTypes is now its own package. We have started to use it for new code.

Reply:
Yes, please, use `PropTypes` from its own package because React 16 is in RC2 now and we'll get a lot of `React.PropTypes` errors eventually (hopefully soon).